.github/workflows/release.yml

```diff
@@ -22,14 +22,121 @@ jobs:
           echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
           nix-shell --pure --run "echo" ./shell.nix
 
-      - name: Remove prerelease label from dependencies
+      - name: Set Image Org
+        run: |
+          [ -z "${{ secrets.IMAGE_ORG }}" ] && IMAGE_ORG=openebs || IMAGE_ORG=${{ secrets.IMAGE_ORG }}
+          echo "IMAGE_ORG=${IMAGE_ORG}" >> $GITHUB_ENV
+
+      - name: Set TAG from github ref name
         run: |
-          nix-shell --pure --run "./scripts/helm/update-version-upon-release.sh" ./shell.nix
+          TAG="${{ github.ref_name }}"
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Mirror images from dev to Docker Hub
+        run: |
+          nix-shell --pure --run "./scripts/staging/mirror-images.sh \
+            --source-namespace ghcr.io/${{ env.IMAGE_ORG }}/dev \
+            --target-namespace docker.io/${{ secrets.DOCKERHUB_USERNAME }} \
+            --tag ${TAG}" ./shell.nix
+
+  release-kubectl-binaries:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: cachix/install-nix-action@v31.3.0
+      - name: Pre-populate nix-shell
+        run: |
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --pure --run "echo" ./shell.nix
```

Comment:
Here and above, should we use another shell.nix? We don't need the rust compiler and all here

Reply:
Indeed. Let me add that.